File: backend/app/models/referral.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.db.base import Base, TimestampMixin

class Referral(TimestampMixin, Base):
    __tablename__ = "referrals"

    id = Column(Integer, primary_key=True, autoincrement=True)
    referrer_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    referred_email = Column(String(255), nullable=True)  # nullable - filled when user registers
    referral_code = Column(String(50), nullable=False, index=True)
    status = Column(String(50), default="pending")  # pending, registered, completed
    
    referrer = relationship("User", foreign_keys=[referrer_id])

File: frontend/src/types/referral.ts
export interface ReferralRecord {
    email: string | null;
    status: 'pending' | 'registered' | 'completed' | string;
    date: string;
}

export interface ReferralInfo {
    referral_code: string;
    referral_link: string;
    referrals: ReferralRecord[];
}

export interface ReferralInviteResponse {
    message: string;
    success: boolean;
    referral_link: string;
}
